WcaOnRails/app/models/competition.rb

```diff
@@ -1729,37 +1729,50 @@ def update_persons_wcif!(wcif_persons, current_user)
       :registration_competition_events,
     ]
     competition_activities = all_activities
+    new_assignments = []
+    removed_assignments = []
     wcif_persons.each do |wcif_person|
+      local_assignments = []
       registration = registrations.find { |reg| reg.user_id == wcif_person["wcaUserId"] }
+      next unless registration
       # NOTE: person doesn't necessarily have corresponding registration (e.g. registratinless organizer/delegate).
-      if registration && wcif_person["roles"]
+      if wcif_person["roles"]
         roles = wcif_person["roles"] - ["delegate", "trainee-delegate", "organizer"] # These three are added on the fly.
         registration.update!(roles: roles)
       end
-      if registration && wcif_person["assignments"]
-        registration.assignments = wcif_person["assignments"].map do |assignment_wcif|
-          existing_assignment = registration.assignments.find do |assignment|
-            assignment.wcif_equal?(assignment_wcif)
+      if wcif_person["assignments"]
+        wcif_person["assignments"].map do |assignment_wcif|
```

Comment:
Nitpick: You're not using the result of the `map` invocation, so `my_array.each do |assignment_wcif|` should do.

Reply:
Fixed!